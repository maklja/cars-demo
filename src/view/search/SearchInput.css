.search-input {
	display: flex;
	justify-content: center;
}

.search-input .search-input__input {
	max-width: 30em;
	border-top-right-radius: 0;
	border-bottom-right-radius: 0;
}

.search-input .search-input__button {
	width: 3em;
	border-top-left-radius: 0;
	border-bottom-left-radius: 0;
	margin-left: 0;
}
