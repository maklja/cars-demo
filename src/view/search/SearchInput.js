import PropTypes from 'prop-types';
import React, { Component } from 'react';

import './SearchInput.css';

export default class SearchInput extends Component {
	constructor(props) {
		super(props);

		this.state = {
			value: ''
		};

		this._onChange = this._onChange.bind(this);
		this._onKeyPress = this._onKeyPress.bind(this);
		this._onSearch = this._onSearch.bind(this);
	}

	render() {
		const { placeholder } = this.props;
		const { value } = this.state;

		return (
			<div className="search-input">
				<input
					placeholder={placeholder}
					className="control search-input__input"
					type="text"
					onChange={this._onChange}
					onKeyPress={this._onKeyPress}
					value={value}
				/>
				<button
					className="button-control search-input__button"
					onClick={this._onSearch}
				>
					<i className="icon-search" />
				</button>
			</div>
		);
	}

	_onKeyPress(e) {
		if (e.key === 'Enter') {
			this._onSearch();
		}
	}

	_onSearch() {
		const { value } = this.state;

		this.props.onSearchCallback(value);
	}

	_onChange(e) {
		this.setState({
			value: e.target.value
		});
	}
}

SearchInput.propTypes = {
	onSearchCallback: PropTypes.func,
	placeholder: PropTypes.string
};

SearchInput.defaultProps = {
	onSearchCallback: () => {},
	placeholder: ''
};
