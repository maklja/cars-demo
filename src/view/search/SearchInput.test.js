import React from 'react';
import renderer from 'react-test-renderer';
import { shallow, configure } from 'enzyme'; // use enzyme to simulate user interaction
import Adapter from 'enzyme-adapter-react-16';

import SearchInput from './SearchInput';

// configure enzyme for react 16
configure({ adapter: new Adapter() });

it('SearchInput renders correctly with default props', () => {
	const tree = renderer.create(<SearchInput />).toJSON();
	expect(tree).toMatchSnapshot();
});

it('SearchInput renders correctly with placeholder', () => {
	const tree = renderer
		.create(<SearchInput placeholder="Test placeholder" />)
		.toJSON();
	expect(tree).toMatchSnapshot();
});

it('set input value and invoke search with button click', done => {
	const inputValue = 'Hello';
	// create shallow copy with enzyme
	const wrapper = shallow(
		<SearchInput
			onSearchCallback={search => {
				// when we click on search button this
				// callback need to invoked with input value
				expect(search).toBe(inputValue);
				done();
			}}
		/>
	);

	// get search input
	const searchInput = wrapper.find('input');
	// simulate change of input
	searchInput.simulate('change', { target: { value: inputValue } });
	// take search button
	const searchButton = wrapper.find('button');
	// and then simulate button click to invoke search callback
	searchButton.simulate('click');
});

it('set input value and invoke search with enter key', done => {
	const inputValue = 'Hello';
	// create shallow copy with enzyme
	const wrapper = shallow(
		<SearchInput
			onSearchCallback={search => {
				// when we click on search button this
				// callback need to invoked with input value
				expect(search).toBe(inputValue);
				done();
			}}
		/>
	);

	// get search input
	const searchInput = wrapper.find('input');
	// simulate change of input
	searchInput.simulate('change', { target: { value: inputValue } });
	// simulate keypress event with key enter
	searchInput.simulate('keypress', { key: 'Enter' });
});
