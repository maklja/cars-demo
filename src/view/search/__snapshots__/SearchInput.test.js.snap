// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput renders correctly with default props 1`] = `
<div
  className="search-input"
>
  <input
    className="control search-input__input"
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder=""
    type="text"
    value=""
  />
  <button
    className="button-control search-input__button"
    onClick={[Function]}
  >
    <i
      className="icon-search"
    />
  </button>
</div>
`;

exports[`SearchInput renders correctly with placeholder 1`] = `
<div
  className="search-input"
>
  <input
    className="control search-input__input"
    onChange={[Function]}
    onKeyPress={[Function]}
    placeholder="Test placeholder"
    type="text"
    value=""
  />
  <button
    className="button-control search-input__button"
    onClick={[Function]}
  >
    <i
      className="icon-search"
    />
  </button>
</div>
`;
